docs/resources/cluster.md

```diff
@@ -45,6 +46,19 @@ resource "cockroach_cluster" "serverless" {
   ]
   delete_protection = false
 }
+
+resource "cockroach_cluster" "basic" {
```

Comment:
Is it a good practice to have more than one resource in the same template?

Separately, what's the significance of the resource name vs the cluster name? Is there anything special that happens depending on what you call the resource, or only the plan field within the resource?

What does it mean if the serverless field is empty? It seems almost like there should be instead a "dedicated" field that is only true for Advanced. WDYT?

Reply:
> Is it a good practice to have more than one resource in the same template?
I don't know the convention. However, this is how it was before; I just added one additional example to the 2 that already existed in `examples/resources/cockroach_cluster/resource.tf`. This doc file is generated from that file, so I assume that if you want to show multiple examples in the `cluster.md` docs page, you need to include multiple examples in the `resource.tf` file.

> Separately, what's the significance of the resource name vs the cluster name? Is there anything special that happens depending on what you call the resource, or only the plan field within the resource?
`cockroach_cluster` is the type of the resource, `basic` is the name of this particular instance of the resource. The resource name is used if you need to reference this resource instance in another TF rule, e.g.:
```
resource "cockroach_client_ca_cert" "example" {
  id            = cockroach_cluster.basic.id
  x509_pem_cert = file("client-ca.crt")
}
```
The resource name is used by TF. The cluster name is used by CC.

> What does it mean if the serverless field is empty?
The `serverless` field needs to be defined for "Basic" and "Standard" clusters. For "Standard" clusters, it needs to contain the `usage_limits {provisioned_vcpus=X}` field. For "Basic" clusters, if no usage limits are specified, then an "Unlimited" cluster is created, where there's no limit to the number of RU's that can be consumed in a given month.

> It seems almost like there should be instead a "dedicated" field that is only true for Advanced. WDYT?
I'm not sure I understand what you're suggesting. Can you restate?